### PR TITLE
Update to 20w21a and DFU v3

### PR DESCRIPTION
**Pull request by Juuxel**
Mojang reworked DynamicOps (with major improvements :heart:) in DFU v3, so JanksonOps had to be half-rewritten.

I also bumped major version to 3 because it's a breaking change because of the DFU update.

Edit: also adds (de)serialization for `EntityAttribute` via the registry

### CODE REVIEW

File: src/main/java/io/github/cottonmc/jankson/JanksonFactory.java [hunk 1]
```diff
@@ -71,24 +70,21 @@
...
 		register(builder, Biome.class,              Registry.BIOME);
-		register(builder, BiomeSourceType.class,    Registry.BIOME_SOURCE_TYPE);
 		register(builder, BlockEntityType.class,    Registry.BLOCK_ENTITY_TYPE);
 		register(builder, Carver.class,             Registry.CARVER);
 		register(builder, ChunkStatus.class,        Registry.CHUNK_STATUS);
 		register(builder, ScreenHandlerType.class,  Registry.SCREEN_HANDLER);
 		register(builder, Decorator.class,          Registry.DECORATOR);
-		register(builder, DimensionType.class,      Registry.DIMENSION_TYPE);
-		
+		register(builder, EntityAttribute.class,    Registry.ATTRIBUTES);
```
Comment (LemmaEOF) [inline src/main/java/io/github/cottonmc/jankson/JanksonFactory.java]: don't forget to register a deserializer for this

**Review by LemmaEOF — CHANGES_REQUESTED** [review]
almost there, just needs the deserializer for entity attributes

File: src/main/java/io/github/cottonmc/jankson/JanksonFactory.java (hunk 1 above)
Comment (LemmaEOF) [reply src/main/java/io/github/cottonmc/jankson/JanksonFactory.java]: actually, wait, this class is kinda weird, just a sec

Comment (Juuxel) [reply src/main/java/io/github/cottonmc/jankson/JanksonFactory.java]: `register` registers both a deserializer and a serializer.

**Review by LemmaEOF — APPROVED** [review]

**Pull request merged**